src/ids-upload-advanced/ids-upload-advanced-file.js

```diff
@@ -0,0 +1,430 @@
+    if (this.status === shared.STATUS.inProcess) {
+      this.status = shared.STATUS.aborted;
+      this.dispatchChangeEvent('abort', e);
+    }
+  }
+
+  /**
+   * Progress handler
+   * @param  {any} e The event
+   * @returns {void}
+   */
+  progressHandler(e) {
+    this.value = (e.loaded / e.total) * 100;
+  }
+
+  /**
+   * Complete handler
+   * @param  {any} e The event
+   * @returns {void}
+   */
+  completeHandler(e) {
+    if (e.target.readyState === 4 && e.target.status === 200) {
+      this.value = '100';
+      this.status = shared.STATUS.completed;
+      this.setStatus();
+    } else {
+      if (this.value === '100') {
+        this.value = '0';
+      }
+      this.errorHandler(e);
+    }
+  }
+
+  /**
+   * Error handler
+   * @param  {any} e The event
+   * @returns {void}
+   */
+  errorHandler(e) {
+    let err = 'Failed to upload, server issue';
+    if (typeof e === 'string' && e !== '') {
+      err = e;
+    } else if (typeof e === 'object') {
+      err = `${e.target.status} - ${e.target.statusText}`;
+    }
+    this.status = shared.STATUS.error;
+    this.error = err;
+  }
+
+  /**
+   * Get loaded value
+   * @private
+   * @returns {number} The close button text
+   */
+  get loaded() {
+    const percent = stringUtils.stringToNumber(this.value);
+    const total = stringUtils.stringToNumber(this.size);
+    return (percent * total) / 100;
+  }
+
+  /**
+   * Get formatted loaded value
+   * @private
+   * @returns {string} The close button text
+   */
+  get loadedFormatted() { return shared.formatBytes(this.loaded); }
+
+  /**
+   * Get formatted size value
+   * @private
+   * @returns {string} The close button text
+   */
+  get sizeFormatted() {
+    return shared.formatBytes(stringUtils.stringToNumber(this.size));
+  }
+
+  /**
+   * Get text for close button
+   * @private
+   * @returns {string} The close button text
+   */
+  get closeButtonText() {
+    let text = shared.slotVal(this.shadowRoot, 'text-btn-cancel');
+    if (this.status === shared.STATUS.errored) {
+      text = shared.slotVal(this.shadowRoot, 'text-btn-close-error');
+    } else if (this.status === shared.STATUS.completed) {
+      text = shared.slotVal(this.shadowRoot, 'text-btn-remove');
+    }
+    return text;
+  }
+
+  /**
+   * Get text for progress label
+   * @private
+   * @returns {string} The progress label text
+   */
+  get progressLabelText() {
+    return shared.slotVal(this.shadowRoot, 'text-progress-label')
+      .replace('{file-name}', this.fileName)
+      .replace('{loaded}', this.loadedFormatted.toString())
+      .replace('{size}', this.sizeFormatted.toString())
+      .replace('{percent}', this.value);
+  }
+
+  /**
+   * Get error html
+   * @private
+   * @returns {string} The error
+   */
+  get errorHtml() {
+    const isInSlot = Object.values(shared.ERRORS).indexOf(this.error) > -1;
+    return isInSlot ? shared.slotVal(this.shadowRoot, this.error) : this.error;
+  }
+
+  /**
+   * Sets to disabled
```

Comment:
```suggestion
   * Sets to disabled
```

Sets what to disabled? I would explain a bit more since this is public